src/TCPDesktop.java

```diff
@@ -0,0 +1,186 @@
+//        this.serverPortNumber = serverPortNumber;
+//        this.socket = socket;
+//        
+//        System.out.println("Start of desktop");
+//        System.out.println("Attempting to connect to server ...");
+//        SSLSession session  = socket.getSession();
+//        
+//        System.setProperty("javax.net.ssl.trustStore", "ca.store");
+//        
+//        oStream = new ObjectOutputStream(socket.getOutputStream());
+//        iStream = new ObjectInputStream(socket.getInputStream());
+//        
+//        socket.getUseClientMode();
+//        
+//        sslSocketFactory.createSocket(socket, iStream, true);
+//    }
+//    
+//    public static void main(String[] args) throws IOException
+//    {
+//        
+//    }
+//}
+import static com.sun.org.apache.bcel.internal.util.SecuritySupport.getResourceAsStream;
+import java.io.BufferedInputStream;
+import java.io.ByteArrayInputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.net.Socket;
+import java.security.KeyStore;
+import java.security.SecureRandom;
+import java.security.Security;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.List;
+import javax.naming.Context;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManagerFactory;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+
+/**
+ * Created by BenBody on 11/05/2017.
+ */
+// Class which handles sending and receiving of data
+public class TCPDesktop
+{
+    private SSLSocket socket;
+    private Socket normalSocket;
+    private ObjectInputStream in;
+    private ObjectOutputStream out;
+    // URL must be changed to the correct URL for the computer the server is running on
+    private static final String SERVERIP = "192.168.1.48"; //These could potentially be read from file, or hard coded
+    private static final int SERVERPORT = 1234; //or found in some other method
+    private static final String KEYSTOREPASS = "capita123"; // same applies to this
+    private boolean isSetUp; // whether the client has been successfully set up
+
+
+    public boolean isSetUp()
+    {
+        return isSetUp;
+    }
+
+    public TCPDesktop()
+    {
+        //set up socket & streams
+        try
+        {
+            // unclear whether this section is required or indeed works
+            // Loads in the keystore from the resources
+            System.out.println("starting setup");
+            Security.addProvider(new BouncyCastleProvider());
+            CertificateFactory cf = CertificateFactory.getInstance("X.509", "BC");
+            InputStream caInput = getResourceAsStream("capita.cer");
```

Comment:
I don think you need to use a certificate as long as you have the keystore, like u have below